### PR TITLE
Create icon sprite consisting of SVG symbols

### PR DESCRIPTION
**Pull request by jonalport**
As we discussed at the FE summit, adding an SVG sprite to house all the icons.
@benbarnett @Guntrisoft

### CODE REVIEW

File: app/views/shared/svg/_use_icon.html.erb [hunk 1]
```diff
@@ -1,3 +1,3 @@
-<svg xmlns="http://www.w3.org/2000/svg" class="svg-icon svg-icon--<%= icon %><%= defined?(variant) ? " svg-icon--#{icon}--#{variant}" : '' %>">
+<svg xmlns="http://www.w3.org/2000/svg" class="svg-icon svg-icon--<%= icon %><%= defined?(variant) ? " svg-icon--#{icon}--#{variant}" : '' %><%= defined?(class_name) ? " #{class_name}" : '' %>">
```
Comment (Guntrisoft) [inline app/views/shared/svg/_use_icon.html.erb]: I was told by Tomas to use local_assigns.has_key?

http://stackoverflow.com/questions/2060561/optional-local-variables-in-rails-partial-templates-how-do-i-get-out-of-the-de

File: app/views/shared/svg/_icon_sprite.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <symbol id="svg-icon--plus" viewBox="0 0 22.354 22.354">
+    <polygon points="8.61,22.354 13.744,22.354 13.744,13.744 22.354,13.744 22.354,8.611 13.744,8.611 13.744,0 8.61,0 8.61,8.611 0,8.611 0,13.744 8.61,13.744" />
+  </symbol>
+  <symbol id="svg-icon--minus" viewBox="0 0 22.354 22.354">
+    <rect y="8.61" width="22.354" height="5.133" />
+  </symbol>
+  <symbol id="svg-icon--arrow-right" viewBox="0 0 22 34">
+    <polygon points="0,5 5,0 22,17 5.1,34 0,29 12,17" />
+  </symbol>
+  <symbol id="icon-arrow-right" viewBox="0 0 22 34">
```
Comment (benbarnett) [inline app/views/shared/svg/_icon_sprite.html.erb]: Is this a deliberate dupe?

File: app/views/shared/svg/_use_icon.html.erb (hunk 1 above)
Comment (darokel) [inline app/views/shared/svg/_use_icon.html.erb]: `local_assigns.fetch :class_name, ' '` would be nicer.

Comment (jonalport) [inline app/views/shared/svg/_use_icon.html.erb]: Yep will update, hadn't heard of these

File: app/views/shared/svg/_icon_sprite.html.erb (hunk 2 above)
Comment (jonalport) [inline app/views/shared/svg/_icon_sprite.html.erb]: Yeah - scared of inadvertently breaking something as "icon-arrow-right" is being referenced elsewhere.

Comment (benbarnett) [inline app/views/shared/svg/_icon_sprite.html.erb]: You can search it across all the repos – would be better to clean this up now tbh. Something like this: https://github.com/search?utf8=%E2%9C%93&q=user%3Amoneyadviceservice+%22%23icon-arrow-right+use%22&type=Code&ref=searchresults

File: app/views/shared/svg/_use_icon.html.erb (hunk 1 above)
Comment (jonalport) [inline app/views/shared/svg/_use_icon.html.erb]: Have updated to use local_assigns

File: app/views/shared/svg/_icon_sprite.html.erb (hunk 2 above)
Comment (jonalport) [inline app/views/shared/svg/_icon_sprite.html.erb]: Yeah, all done

**Comment by benbarnett** [comment]
Pretty sweet :+1:

**Comment by asmega** [comment]
:+1:

**Pull request merged**